torch/onnx/_internal/diagnostics/infra/context.py

```diff
@@ -96,13 +104,87 @@ def with_graph(self: _Diagnostic, graph: infra.Graph) -> _Diagnostic:
         self.graphs.append(graph)
         return self
 
-    def with_additional_message(self: _Diagnostic, message: str) -> _Diagnostic:
-        """Adds an additional message to the diagnostic."""
-        if self.additional_message is None:
-            self.additional_message = message
-        else:
-            self.additional_message = f"{self.additional_message}\n{message}"
-        return self
+    @contextlib.contextmanager
+    def log_section(
```

Comment:
```suggestion
    def log_context(
```
more aligned with the "context manager" concept in python

What is the difference between `diagnostic_context` and `log_section`? Could one be part of the other and publicly the user only care about the only public API?

Reply:
this is not what it meant. I'll add examples.